components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/api/impl/BusinessRulesApiServiceImpl.java

```diff
@@ -399,34 +400,61 @@ public Response loadDeploymentInfo(Request request, String businessRuleInstanceI
     }
 
     @Override
-    public Response redeployBusinessRule(Request request, String businessRuleInstanceID) throws NotFoundException {
+    public Response deployOrUndeployBusinessRule(Request request, String businessRuleInstanceID,
+                                                 boolean shouldUndeploy) throws NotFoundException {
         if (!hasPermission(request, RequestMethod.REDEPLOY_BUSINESS_RULE)) {
             return Response.status(Response.Status.UNAUTHORIZED).build();
         }
+
         TemplateManagerService templateManagerService = TemplateManagerInstance.getInstance();
         Gson gson = new GsonBuilder().disableHtmlEscaping().setPrettyPrinting().create();
         List<Object> responseData = new ArrayList<Object>();
         try {
-            int status = templateManagerService.redeployBusinessRule(businessRuleInstanceID);
+
+            status = templateManagerService.deployOrUndeployBusinessRule(businessRuleInstanceID, shouldUndeploy);
+
             switch (status) {
                 case (TemplateManagerConstants.DEPLOYED):
                     responseData.add("Deployment Successful");
                     responseData.add("Successfully deployed the business rule");
                     break;
-                case (TemplateManagerConstants.PARTIALLY_UNDEPLOYED):
+                case (TemplateManagerConstants.SAVED):
+                    responseData.add("Deployment Successful");
+                    responseData.add("Successfully undeployed the business rule");
```

Comment:
`responseData.add("Deployment Successful");`
if the status is saved, why we are saying deployment successful?

`responseData.add("Successfully undeployed the business rule");`
These messages are not correct. Plz re-check and correct them.

Reply:
I will check and do the changes

Reply:
I fixed this changes